# Pair Programming Estadística I

1. Buscar un conjunto de datos a analizar:
- Se recomienda que el conjunto de datos a analizar tenga variables numéricas y categóricas. Ya que también se analizarán.

In [2]:
import pandas as pd
import numpy as np
import pickle
import os

In [3]:
os.getcwd()

'/mnt/c/Users/penny/OneDrive/Documentos/ADALAB/pair programming/promo-d-da-modulo2-sprint1-laura-paula-estibaliz/estadistica'

In [3]:
df_train = pd.read_csv("../datos/train.csv")
df_train.head()

Id  MSSubClass MSZoning  LotFrontage  LotArea Street Alley LotShape  \
0   1          60       RL         65.0     8450   Pave   NaN      Reg   
1   2          20       RL         80.0     9600   Pave   NaN      Reg   
2   3          60       RL         68.0    11250   Pave   NaN      IR1   
3   4          70       RL         60.0     9550   Pave   NaN      IR1   
4   5          60       RL         84.0    14260   Pave   NaN      IR1   

  LandContour Utilities  ... PoolArea PoolQC Fence MiscFeature MiscVal MoSold  \
0         Lvl    AllPub  ...        0    NaN   NaN         NaN       0      2   
1         Lvl    AllPub  ...        0    NaN   NaN         NaN       0      5   
2         Lvl    AllPub  ...        0    NaN   NaN         NaN       0      9   
3         Lvl    AllPub  ...        0    NaN   NaN         NaN       0      2   
4         Lvl    AllPub  ...        0    NaN   NaN         NaN       0     12   

  YrSold  SaleType  SaleCondition  SalePrice  
0   2008        WD         Normal     208500  
1   2007        WD         Normal     181500  
2   2008        WD         Normal     223500  
3   2006        WD        Abnorml     140000  
4   2008        WD         Normal     250000  

[5 rows x 81 columns]

In [4]:
homogeneizar_columnas = {col: col.lower().replace(" ", "_") for col in df_train}
df_train.rename(columns = homogeneizar_columnas, inplace = True)
df_train.head()

id  mssubclass mszoning  lotfrontage  lotarea street alley lotshape  \
0   1          60       RL         65.0     8450   Pave   NaN      Reg   
1   2          20       RL         80.0     9600   Pave   NaN      Reg   
2   3          60       RL         68.0    11250   Pave   NaN      IR1   
3   4          70       RL         60.0     9550   Pave   NaN      IR1   
4   5          60       RL         84.0    14260   Pave   NaN      IR1   

  landcontour utilities  ... poolarea poolqc fence miscfeature miscval mosold  \
0         Lvl    AllPub  ...        0    NaN   NaN         NaN       0      2   
1         Lvl    AllPub  ...        0    NaN   NaN         NaN       0      5   
2         Lvl    AllPub  ...        0    NaN   NaN         NaN       0      9   
3         Lvl    AllPub  ...        0    NaN   NaN         NaN       0      2   
4         Lvl    AllPub  ...        0    NaN   NaN         NaN       0     12   

  yrsold  saletype  salecondition  saleprice  
0   2008        WD         Normal     208500  
1   2007        WD         Normal     181500  
2   2008        WD         Normal     223500  
3   2006        WD        Abnorml     140000  
4   2008        WD         Normal     250000  

[5 rows x 81 columns]

2. Extraed los siguentes estadísiticos del conjunto de datos para varias de las variables disponibles:

### Columna 'Saleprice' - Variable 'Precio de venta'

1. Media

In [9]:
media = round(df_train["saleprice"].mean(), 3)
f'La media de la columna "saleprice" es {media}'

'La media de la columna "saleprice" es 180921.196'

2. Desviación con respecto de la media

In [13]:
diferencias = df_train["saleprice"] - media
print(diferencias)

0       27578.80411
1         578.80411
2       42578.80411
3      -40921.19589
4       69078.80411
           ...     
1455    -5921.19589
1456    29078.80411
1457    85578.80411
1458   -38796.19589
1459   -33421.19589
Name: saleprice, Length: 1460, dtype: float64


In [14]:
desviacion_media = np.abs(diferencias)
print(desviacion_media)

0       27578.80411
1         578.80411
2       42578.80411
3       40921.19589
4       69078.80411
           ...     
1455     5921.19589
1456    29078.80411
1457    85578.80411
1458    38796.19589
1459    33421.19589
Name: saleprice, Length: 1460, dtype: float64


3. Varianza

In [15]:
varianza = df_train["saleprice"].var()
f'La varianza de la columna "saleprice" es {varianza}'

'La varianza de la columna "saleprice" es 6311111264.297451'

4. Desviación típica

In [10]:
des_tipica = round(df_train["saleprice"].std(), 3)
f'La desviación estándar de la columna "saleprice" es {des_tipica}'

'La desviación estándar de la columna "saleprice" es 79442.503'

5. Moda

In [17]:
moda = df_train["saleprice"].mode()[0]
f'La moda de la columna "saleprice" es {moda}'

'La moda de la columna "saleprice" es 140000'

6. Mediana

In [18]:
mediana = df_train["saleprice"].median()
f'La mediana de la columna "saleprice" es {mediana}'

'La mediana de la columna "saleprice" es 163000.0'

7. Media ponderada

In [25]:
w = np.random.randint(0,100, df_train.shape[0])

avg = np.average(df_train["saleprice"], weights=w)

f'La media ponderada de la columna "saleprice" es {avg}'


'La media ponderada de la columna "saleprice" es 181184.2571790713'

8. Robustez

- Temenos 1460 entradas (filas), que nos parece una buena amuestra en términos de tamaño. Este número nos sugere que por este lado los datos son robustos.
- La media de la columna "salesprice" es 180921.196 y la deviación estandard es 79442.503, o sea, la deviacíon estandar es bastanta alta lo que hace con que tengamos muchos dados que se alejan de la media. Eso nos hace con que la media no refleje de forma adecuada el valor de la mayoria de los datos que tenemos en nuestra columna. 

9. Coeficiente de variación

In [19]:
coef_var = df_train["saleprice"].std()/df_train["saleprice"].mean()
f'El coeficiente de variación de la columna "saleprice" es {coef_var}'

'El coeficiente de variación de la columna "saleprice" es 0.4391000318780072'

### Columna 'Lot Area'

In [23]:
media2 = df_train['lotarea'].mean()
print(f'La media de la columna "LotArea" es {media2}')
print('----------')

mediana2 = df_train['lotarea'].median()
print(f'La mediana de la columna "LotArea" es {mediana2}')
print('----------')

moda2 = df_train['lotarea'].mode()[0]
print(f'La moda de la columna "LotArea" es {moda2}')
print('----------')

maximo = df_train['lotarea'].max()
print(f'El valor máximo de la columna "LotArea" es {maximo}')
print('----------')

minimo = df_train['lotarea'].min()
print(f'El valor mínimo de la columna "LotArea" es {minimo}')
print('----------')

des_tipica2 = df_train['lotarea'].std()
print(f'La desviación típica de la columna "LotArea" es {des_tipica2}')
print('----------')

varianza2 = df_train['lotarea'].var()
print(f'La varianza de la columna "LotArea" es {varianza2}')
print('----------')

coef_var2 = df_train['lotarea'].std()/df_train['lotarea'].mean()
print(f'El coeficiente de variación de la columna "LotArea" es {coef_var2}')
print('----------')


La media de la columna "LotArea" es 10516.828082191782
----------
La mediana de la columna "LotArea" es 9478.5
----------
La moda de la columna "LotArea" es 7200
----------
El valor máximo de la columna "LotArea" es 215245
----------
El valor mínimo de la columna "LotArea" es 1300
----------
La desviación típica de la columna "LotArea" es 9981.264932379147
----------
La varianza de la columna "LotArea" es 99625649.6503417
----------
El coeficiente de variación de la columna "LotArea" es 0.9490756009676047
----------


- Desviación respecto a la media

In [24]:
diferencias2 = df_train["lotarea"] - media
print(diferencias2)

desviacion_media2 = np.abs(diferencias2)
print(f' La desviación respecto a la media de la columna "LotArea" es {desviacion_media2}')

0      -172471.19589
1      -171321.19589
2      -169671.19589
3      -171371.19589
4      -166661.19589
            ...     
1455   -173004.19589
1456   -167746.19589
1457   -171879.19589
1458   -171204.19589
1459   -170984.19589
Name: lotarea, Length: 1460, dtype: float64
 La desviación respecto a la media de la columna "LotArea" es 0       172471.19589
1       171321.19589
2       169671.19589
3       171371.19589
4       166661.19589
            ...     
1455    173004.19589
1456    167746.19589
1457    171879.19589
1458    171204.19589
1459    170984.19589
Name: lotarea, Length: 1460, dtype: float64


- Media ponderada

In [26]:
w2 = np.random.randint(0,100, df_train.shape[0])

avg2 = np.average(df_train["lotarea"], weights=w2)

f'La media ponderada de la columna "lotarea" es {avg2}'

'La media ponderada de la columna "lotarea" es 10376.339728273932'

3. Interpretación de los resultados.

Ahora interpreta los resultados obtenidos de los diferentes estadísticos estudiados haciendo hincapié en aspectos interesantes de tus datos y explica lo mejor posible la naturaleza de los mismos.

- Temenos 1460 entradas (filas), que nos parece una buena amuestra en términos de tamaño. Este número nos sugere que por este lado los datos son robustos.
- La media de la columna "lotarea" es 10516.82 y la deviación estandard es 9981.26, o sea, la deviacíon estandar es bastanta alta (casi llega al mismo valor de la media) lo que hace con que tengamos muchos dados que se alejan de la media. Eso nos hace con que la media no refleje de forma adecuada el valor de la mayoria de los datos que tenemos en nuestra columna. 